### DISCUSSION TITLE
Can Llama-cpp-python caches be saved to disk and later reloaded during inference?

### DISCUSSION THREAD

**Discussion by yentur** [abetlen/llama-cpp-python#1142] [Q&A]
Is it possible to preprocess a set of prompts, save them to disk, and then reload them during inference to avoid the need to regenerate cached prompts every time? For instance, if I have a 2000-token prompt that I use daily in a memory-intensive Python program, is there a way to pre-process and save it to avoid the delay of ingesting the prompt each time I start the program? What are the options in this scenario?

**Comment by yamikumo-DSD** [comment]
Not sure you're still interested in the issue, but LlamaDiskCache does the job, even if it has a slight bug now.
